lib/Ethereum/RPC/Contract.pm

```diff
@@ -66,8 +67,11 @@ sub BUILD {
 
     my @decoded_json = @{decode_json($self->contract_abi)};
 
-    foreach my $json_input (@decoded_json) {
-        $self->contract_decoded->{$json_input->{name}} = \@{$json_input->{inputs}} if $json_input->{type}  eq 'function';
+    for my $json_input (@decoded_json) {
+        if ( $json_input->{type}  eq 'function' ) {
+            $self->contract_decoded->{$json_input->{name}} = [] unless $self->contract_decoded->{$json_input->{name}};
+            push(@{$self->contract_decoded->{$json_input->{name}}}, \@{$json_input->{inputs}}) if scalar @{$json_input->{inputs}} > 0;
```

Comment:
`\@{$json_input->{inputs}}` is somewhat odd, unless you truly want to make a new copy of the arrayref then make a new ref out of that copy.